### ISSUE TITLE
Section on CBC padding attack does not mention IV

### ISSUE THREAD

**Issue by dato** [crypto101/book#352]
I’ve used section 7.9, _CBC padding attacks_ as a guide to complete [challenge #17](https://cryptopals.com/sets/3/challenges/17) of the cryptopals challenges.

Section 7.9 is a _great_ write-up, but to my frustration I couldn’t make the attack work. Turns out the “Decrypting one byte” subsection should really have made a reference to the initialization vector (IV), because it’s needed to compute the _actual_ byte.

In other words, in the formula on page 70, the expression `01 ⊕ r[b]` should be, as I understand it, `01 ⊕ r[b] ⊕ iv[b]`.

Other than that thanks for the excellent description.

(FWIW the challenge in cryptopals.com does mention “oracle function returns the ciphertext and IV”, but I quickly forgot about it when following the book.)

**Comment by dato** [comment]
Oh.

> This is sorta intentional.

Thanks for the explanation. I think I understand your point. What I would like now is, with your explanation as starting point, to restate my original observation and see if the text could still be improved.

>Does the expression make sense with the drawing?

So, the formula I complained about seem to me now, in a narrow sense, correct. When I complained that the formula “only [achieves us] the immediate output of D, not the plaintext”, well, the formula does not promise otherwise. After all, it says `D(Ci)[b]`, and not `Pi[b]`. Is this what you meant?

Still I think it would be great if the text would explain... how to get from `D(Ci)[b]` to `Pi[b]` (and to point to the difference in the first place!). It’s true this can be deduced by comparing the image in §7.9 with the original images for CBC decryption in §7.4; but I can’t think of a reason not to help the reader here.

The explanation on how to get `Pi[b]` is useful even if the IV is not available, since it can be applied to every block except the first one. I think it can be added with no loss of generality? The text would just need to mention the difference between block 0, and the rest, with respect to the availability of the IV. 

Does this at least make sense? “Help the reader” would be my main point.

**Comment by lvh** [comment]
> Thanks for the explanation; I think I get your point. What I’d like to do now is, with your explanation as starting point, restate my original observation and see if the text could still be improved.

Yeah definitely: I didn't mean to sound dismissive of your issue, I do want to fix the text :)

I also agree the distinction between D(Ci) and Pi needs to be highlighted and particularly how you get from one to the other. How do you feel about writing prose?

**Comment by dato** [comment]
>I didn't mean to sound dismissive of your issue.

All is good. :)

>How do you feel about writing prose?

Heh, I can look into if if you’d like. Should have time for it some time before the end of the month.
